packages/firestore/src/lite-api/settings.ts

```diff
@@ -123,17 +123,23 @@ export class FirestoreSettingsImpl {
       }
     }
 
-    this.experimentalForceLongPolling = !!settings.experimentalForceLongPolling;
-    this.experimentalAutoDetectLongPolling =
-      !!settings.experimentalAutoDetectLongPolling;
-    this.useFetchStreams = !!settings.useFetchStreams;
-
     validateIsNotUsedTogether(
       'experimentalForceLongPolling',
       settings.experimentalForceLongPolling,
       'experimentalAutoDetectLongPolling',
       settings.experimentalAutoDetectLongPolling
     );
+
+    if (!settings.experimentalForceLongPolling) {
+      this.experimentalForceLongPolling = false;
+      this.experimentalAutoDetectLongPolling =
+        settings.experimentalAutoDetectLongPolling ?? false;
```

Comment:
I think we should respect previous code that would ensure runtime type is boolean.

I prefer using `Boolean` function instead of `!!`, but you call.

This was helpful to me when considering trade off: https://gist.github.com/arthurvi/66cb1e2bcfc92f99f465e0db04264367

Reply:
TLDR: All this will be very opinionated, so in the end, I am ok with original code and your new code, so I will simply approve. Feel free to make further changes if you want.

Here is the TL part:

Looking at surrounding code and digging a little deeper, I do see many things worth noting.

1. IntelliJ doesn't complain when using `Boolean(..)` instead of `!!`.
2. `DEFAULT_SSL` uses the `??` operator, so if it is a problem here, it will also be a problem there.
3. Handling these corner cases is increasing code size, but adding questionable value. I think there is a trade off to consider.
4. `DEFAULT_SSL` introduces a constant. That pattern, if followed, implies that long polling should also have constants. But this only serves to bloat the code further.
5. I am not sure whether minification will help with any of the above. I suspect not.

Reply:
Thanks for these thoughts, Tom.

I've opened a new PR, https://github.com/firebase/firebase-js-sdk/pull/7242, which expands the unit test suite to also verify how "truthy" and "falsy" values are coerced into `boolean`. PTAL if you don't mind. I wanted to add those tests in a separate PR (rather than including them in this PR) in order to "prove" that there are no behavior changes introduced by this PR.

To avoid unintentionally introducing any backwards incompatibilities, I'm also going to modify this PR to coerce to boolean. I've also modified the logic to more closely mirror that of `DEFAULT_SSL`, as you pointed out. Namely, I've added a new constant, `DEFAULT_AUTO_DETECT_LONG_POLLING`, and used the `??` operator where appropriate.

Personally, I'm not concerned about the code size increases from changes like this since the delta will be minuscule.